packages/x-adapter/src/utils/interpolate.ts

```diff
@@ -0,0 +1,99 @@
+/**
+ * Syntax to detect and extract string parameters. A string parameter contains a property name
+ * with an optional header or tail to concatenate wrapped in curly braces (`{}`).
+ * The different parts of a string parameter are explained in {@link STRING_PARAMETER_CONTENT}.
+ *
+ * @example Different string parameters
+ * ```js
+ *   "\{env\}" // No optional head or tail.
+ *   "\{(.)env\}" // Optional `.` head.
+ *   "\{env(-api)\}" // Optional `-api` tail.
+ *   "\{(api-)env(.)\}" // Optional `api-` head and `.` tail.
+ * ```
+ * @internal
+ */
+const STRING_PARAMETERS = /{([^}]+)}/g;
+
+/**
+ * Syntax of a single string parameter. A string parameter shape is composed by
+ * the name of the property that should be replaced. This property name can be preceded
+ * or followed by an optional string to prepend or to append to the property value
+ * in case it is defined.
+ *
+ * @example Valid string parameters content
+ * ```js
+ *   "env" // No optional head or tail to concatenate.
+ *   "(.)env" // The `.` character will be prepended as long as the `env` property is defined.
+ *   "env(-api)" // The `-api` string will be appended as long as the `env` property is defined.
+ *   "(api-)env(.)" // The `api-` string and the `.` character will be added as long as
+ *   // the `env` property is defined.
+ * ```
+ * @internal
+ */
+const STRING_PARAMETER_CONTENT = /^(?:\((.+)\))?([^(]+)(?:\((.+)\))?$/g;
```

Comment:
What do you thin about splitting this?

`
const optionalParameter = '(?:\\((.+)\\))?';

const STRING_PARAMETER_CONTENT = new RegExp(
  `^${optionalParameter}([^(]+)${optionalParameter}$`,
  'g'
);
`

Reply:
I prefer the first one because I don't have to repeat that weird code twice, but I don't like the name, and I can't think something better 🤦